### ISSUE TITLE
Configuring clangd on a per-project basis

### ISSUE THREAD

**Issue by tbodt** [clangd/clangd#71]
Some clangd options, such as header insertion policy, would be best configured in a per-project file, rather than in the user's global configuration.

**Comment by sam-mccall** [comment]
Yes. The background indexer writes to $PROJECT/.clangd/index, so something under $PROJECT/.clangd would make sense.

We may also want to move some of the options that are currently flags to a user configuration file.

Open questions: format, backwards compatibility model, which options should be offered where...

**Comment by lh123** [comment]
@sam-mccall  I think this feature only needs to separate `clangd.arguments` into multiple options.

**Comment by kadircet** [comment]
We already have a `Completion` block in config today. I think we just need the wiring of a new option.

**Comment by HighCommander4** [comment]
I believe this has been addressed broadly by the introduction of [clangd config files](https://clangd.llvm.org/config.html), and specifically by the addition of a config key for header insertion in https://github.com/llvm/llvm-project/pull/128503.
